web-client/src/app/services/auto-logout.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable } from '@angular/core';
+import { resetStores } from '@datorama/akita';
+import { NavController } from '@ionic/angular';
+import { SessionStore } from 'src/app/state/session.store';
+import { SessionQuery } from '../state/session.query';
+import { SwalHelper } from '../utils/notification/swal-helper';
+
+const MINUTES_UNTIL_AUTO_LOGOUT = 1; // in mins
+const CHECK_INTERVAL = 5000; // in ms
+const STORE_KEY = 'lastAction';
+@Injectable({
+  providedIn: 'root',
+})
+export class AutoLogoutService {
+  isLogin = false;
+  constructor(
+    private navCtrl: NavController,
+    private sessionStore: SessionStore,
+    public sessionQuery: SessionQuery,
+    private notification: SwalHelper
+  ) {
+    this.check();
+    this.initListener();
+    this.initInterval();
+    localStorage.setItem(STORE_KEY, Date.now().toString());
+    if (this.sessionQuery.isActiveSession()) {
+      this.isLogin = true;
+    }
+  }
+  public getLastAction() {
+    return parseInt(localStorage.getItem(STORE_KEY) || '', 10);
+  }
+  public setLastAction(lastAction: number) {
+    localStorage.setItem(STORE_KEY, lastAction.toString());
+  }
+
+  initListener() {
+    addEventListener('click', () => this.reset());
+    addEventListener('mousemove', () => this.reset());
+    addEventListener('keydown', () => this.reset());
+    addEventListener('touchmove', () => this.reset());
+  }
+
+  reset() {
+    this.setLastAction(Date.now());
+  }
+
+  initInterval() {
+    setInterval(() => {
+      this.check();
+    }, CHECK_INTERVAL);
+  }
+
+  check() {
+    if (this.isLogin) {
+      const now = Date.now();
+      const timeLeft =
+        this.getLastAction() + MINUTES_UNTIL_AUTO_LOGOUT * 60 * 1000;
+      const diff = timeLeft - now;
+      const isTimeout = diff < 0;
+
+      if (isTimeout) {
+        this.cleanUp(true);
+      }
+    }
+  }
+
+  public cleanUp(notification: boolean) {
+    localStorage.removeItem(STORE_KEY);
+    removeEventListener('click', () => this.reset());
```

Comment:
why not `removeAllListeners`?

Reply:
Didn't know there was such a concise method :) thanks!

Reply:
I got some errors when calling `removeAllListeners` so I've reverted it back to `removeEventListener`.